File: deploy-server.py
#!/usr/bin/env python3

import subprocess
import sys

def run_ssh_command(cmd):
    """Run a command on the remote server via SSH"""
    full_cmd = [
        'ssh',
        '-o', 'ConnectTimeout=30',
        '-o', 'StrictHostKeyChecking=no',
        'root@159.65.181.148',
        cmd
    ]
    print(f"Running: {cmd}")
    result = subprocess.run(full_cmd, capture_output=True, text=True, timeout=180)
    print(result.stdout)
    if result.stderr:
        print("STDERR:", result.stderr)
    return result.returncode == 0

try:
    print("=== Deploying fix to production server ===\n")
    
    print("1. Pulling latest code...")
    if not run_ssh_command('cd /var/www/pcs-ui && git pull origin main'):
        print("❌ Git pull failed")
        sys.exit(1)
    
    print("\n2. Rebuilding...")
    if not run_ssh_command('cd /var/www/pcs-ui && npm run build'):
        print("❌ Build failed")
        sys.exit(1)
    
    print("\n3. Restarting PM2...")
    if not run_ssh_command('pm2 restart pcs-ui'):
        print("❌ PM2 restart failed")
        sys.exit(1)
    
    print("\n✅ Deployment complete!")
    sys.exit(0)
    
except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)

File: deploy-server.js
#!/usr/bin/env node

const { execSync } = require('child_process');

function runCommand(cmd, description) {
  console.log(`\n${description}...`);
  try {
    const output = execSync(cmd, { 
      encoding: 'utf-8',
      stdio: 'pipe',
      timeout: 180000
    });
    console.log(output);
    return true;
  } catch (error) {
    console.error(`❌ ${description} failed:`, error.message);
    return false;
  }
}

async function deploy() {
  console.log('=== Deploying fix to production server ===\n');
  
  const commands = [
    {
      cmd: 'ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no root@159.65.181.148 "cd /var/www/pcs-ui && git pull origin main"',
      desc: '1. Pulling latest code'
    },
    {
      cmd: 'ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no root@159.65.181.148 "cd /var/www/pcs-ui && npm run build"',
      desc: '2. Rebuilding'
    },
    {
      cmd: 'ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no root@159.65.181.148 "pm2 restart pcs-ui"',
      desc: '3. Restarting PM2'
    }
  ];
  
  for (const { cmd, desc } of commands) {
    if (!runCommand(cmd, desc)) {
      process.exit(1);
    }
  }
  
  console.log('\n✅ Deployment complete!');
  process.exit(0);
}

deploy().catch(err => {
  console.error('❌ Deployment failed:', err);
  process.exit(1);
});

